pkg/gitserver/controller/controller.go

```diff
@@ -0,0 +1,231 @@
+package controller
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/client/cache"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/gitserver/util"
+)
+
+// BuildConfigController watches build resources and manages their state
+type BuildConfigController struct {
```

Comment:
@kargakis this is not part of the main process. It's created in every git server you launch.